src/metrics.cpp

```diff
@@ -203,16 +203,59 @@ void ConnectMetricsScreen()
     uiInterface.InitMessage.connect(metrics_InitMessage);
 }
 
+std::string DisplayDuration(int64_t time, DurationFormat format)
+{
+    int days =  time / (24 * 60 * 60);
+    int hours = (time - (days * 24 * 60 * 60)) / (60 * 60);
+    int minutes = (time - (((days * 24) + hours) * 60 * 60)) / 60;
+    int seconds = time - (((((days * 24) + hours) * 60) + minutes) * 60);
```

Comment:
No need to change this as what you have is fine, but here is how I like to extract seconds/minutes/hours/etc when dealing with timestamps (in pseudo-code):
```
seconds = time % 60
time /= 60
minutes = time % 60
time /= 60
hours = time % 24
time /= 24
...
```
I find it especially nice doing it that way in languages which have a div-mod operator.

Again, what you have is fine, but thought you might be curious to see the above.

Reply:
I agree with you. That code is actually a c&p of what it was already there in `printMetrics`. I tried to modify the less code possible but your pseudo-code definitely looks better for me. We can maybe open an issue for it.